src/app/core/ui/navigation/menu-item/menu-item.component.ts

```diff
@@ -37,11 +39,36 @@ export class MenuItemComponent {
 
   isExpanded: boolean = false;
 
+  /**
+   * Toggles the submenu visibility.
+   */
   toggleSubMenu(): void {
     this.isExpanded = !this.isExpanded;
   }
 
+  /**
+   * Checks if a menu item has a submenu.
+   */
   hasSubMenu(item: MenuItem): boolean {
     return !!item.subMenu && item.subMenu.length > 0;
   }
+
+  /**
+   * Removes a submenu item at the specified index.
+   */
+  removeSubMenuItem(index: number): void {
+    if (this.item.subMenu && index > -1) {
+      this.item.subMenu.splice(index, 1);
+    }
+  }
+
+  /**
+   * Handles drag-and-drop reordering of submenu items.
+   */
+  dropSubMenu(event: CdkDragDrop<MenuItem[]>): void {
+    if (this.item.subMenu) {
+      moveItemInArray(this.item.subMenu, event.previousIndex, event.currentIndex);
+    }
+  }
```

Comment:
I think these are also not used here, are they?

Maybe go through the changed files here on GitHub to check if the things are still relevant changes